.pre-commit-config.yaml

```diff
@@ -0,0 +1,45 @@
+---
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: (^doc/orig/fosdem2020)
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-case-conflict
+      - id: check-symlinks
+      - id: destroyed-symlinks
+
+      - id: check-json
+      - id: check-yaml
+      - id: check-xml
+
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.0
+    hooks:
+      - id: prettier
+        files: \.(md|yml|yaml|json)$
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        args:
+          - --ignore-words=.codespellignore
+          - --builtin=clear,rare,informal
+
+  - repo: local
+    hooks:
+      - id: ocamlformat
+        name: ocamlformat
+        description: Formatting ML files
+        entry: ocamlformat
```

Comment:
The location of this binary depends on opam env variables. Usually, you run `eval $(opam env)` first to set the proper env variables.